internal/pkg/views/notifications.go

```diff
@@ -0,0 +1,175 @@
+package views
+
+import (
+	"context"
+	"fmt"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/style"
+	"github.com/lawrencegripper/azbrowse/pkg/armclient"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jroimartin/gocui"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/eventing"
+)
+
+func init() {
+	pendingDeletes = []pendingDelete{}
+}
+
+type pendingDelete struct {
+	display string
+	url     string
+}
+
+var pendingDeletes []pendingDelete
+var deleteMutex sync.Mutex // ensure delete occurs only once
+var gui *gocui.Gui
+
+// AddPendingDelete queues deletes for
+// delete once confirmed
+func AddPendingDelete(display, url string) {
+	if url == "" {
+		eventing.SendStatusEvent(eventing.StatusEvent{
+			Failure: true,
+			Message: "Item `" + display + "` doesn't support delete",
+			Timeout: time.Second * 5,
+		})
+		return
+	}
+	gui.Update(func(g *gocui.Gui) error {
+		deleteMutex.Lock()
+		defer deleteMutex.Unlock()
+
+		for _, i := range pendingDeletes {
```

Comment:
I know we talked about this but I've forgotten why this needed to be an ordered list rather than a map?

Reply:
I think it's mainly because I feel it's more intuitive for the list to be ordered so you add something and it goes on the bottom.

Not super computer science wise but hopefully the code is easy to follow - what do you think?